src/wallet/coincontrol.h

```diff
@@ -61,82 +128,153 @@ class CCoinControl
     int m_max_depth = DEFAULT_MAX_DEPTH;
     //! SigningProvider that has pubkeys and scripts to do spend size estimation for external inputs
     FlatSigningProvider m_external_provider;
+    //! Locktime
+    std::optional<uint32_t> m_locktime;
+    //! Version
+    std::optional<uint32_t> m_version;
 
     CCoinControl();
 
     bool HasSelected() const
     {
-        return (setSelected.size() > 0);
+        return (m_selected.size() > 0);
     }
 
     bool IsSelected(const COutPoint& output) const
     {
-        return (setSelected.count(output) > 0);
+        return (m_selected.count(output) > 0);
     }
 
     bool IsExternalSelected(const COutPoint& output) const
     {
-        return (m_external_txouts.count(output) > 0);
+        const auto it = m_selected.find(output);
+        if (it == m_selected.end()) {
+            return false;
+        }
+        return it->second.HasTxOut();
     }
 
     bool GetExternalOutput(const COutPoint& outpoint, CTxOut& txout) const
     {
-        const auto ext_it = m_external_txouts.find(outpoint);
-        if (ext_it == m_external_txouts.end()) {
+        const auto it = m_selected.find(outpoint);
+        if (it == m_selected.end()) {
+            return false;
+        }
+        if (!it->second.HasTxOut()) {
             return false;
         }
-        txout = ext_it->second;
+        txout = it->second.GetTxOut();
         return true;
     }
 
-    void Select(const COutPoint& output)
+    void Select(const COutPoint& output, std::optional<uint32_t> sequence = std::nullopt, std::optional<CScript> script_sig = std::nullopt, std::optional<CScriptWitness> script_witness = std::nullopt)
```

Comment:
I think it would be better to return the `PreselectedInput` and have the caller use `SetSequence` etc directly.

Reply:
Done